dependency_cache.go

```diff
@@ -182,7 +182,22 @@ func (d *DependencyCache) Artifact(dependency BuildpackDependency, mods ...Reque
 			color.New(color.FgYellow, color.Bold).Sprint("Warning:"))
 
 		d.Logger.Bodyf("%s from %s", color.YellowString("Downloading"), uri)
-		artifact = filepath.Join(d.DownloadPath, filepath.Base(uri))
+
+		// ensure query parameters are not included in the downloaded file name if the uri is http type
+		downloadUri, err := url.Parse(uri)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse the download uri %s\n%w", uri, err)
+		}
+
+		if(downloadUri.Scheme == "http" || downloadUri.Scheme == "https") {
```

Comment:
Thinking about this. Should we have an if block here that checks if there are query params? Logically, I believe we only need to do this if there are query params.

Conversely, I'd say we just hash things across the board, but I'd want to check if that would have any side effects before doing that.

Reply:
👍  we could consider it's always http/https and remove this check

Reply:
I looked over the code again and I think it's safe to just use the hasher all the time. That means you don't need to parse the URL, you can just feed the full URL as a string into the hasher and use the hash as the file name.

That said, you also need to do this at https://github.com/paketo-buildpacks/libpak/blob/6a1c758bae92ef4ff1bca7e40c71a65b4140ec74/dependency_cache.go#L236-L240

The code you've modified here is for the case where the dependency has no sha256 in the metadata. If there is a sha256 in the metadata, then you'll skip this whole block and it will download the file on lines 236-240 down below.

Reply:
got it, I'll start working on the changes

Reply:
@dmikusa updated. URI sha will now be used as the downloaded artifact's filename regardless of the value of dependency.SHA256